### PR TITLE
[Winlogbeat] Add configuration for registry file flush timeout

### LINKED ISSUE #29001: [Winlogbeat] Add registry flush settings
**Describe the enhancement:**

Current the interval at which Winlogbeat flushes its registry to disk is not configurable. After 10 updates or 5 seconds it flushes to disk. Under heavy event volumes this can cause an excess of disk and CPU activity. Winlogbeat should offer config option to control the flush interval.

Filebeat uses `filebeat.registry.flush` and it is only based on time. I propose that Winlogbeat follow the same model.

> `winlogbeat.registry.flush: 5s`

**Describe a specific use case for the enhancement or feature:**

This would help limit the amount of CPU and disk IO consumed when under heavy event volumes.

### PR DESCRIPTION
**Pull request by taylor-swanson**
## What does this PR do?

- Added a new option ("winlogbeat.registry_flush") for configuring the
timeout for writting registry entries to disk. The previously hard-coded
value of 5 seconds is now the default value.

## Why is it important?

The value is currently hard-coded and environments that have lots of event activity will cause unnecessary CPU usage due to excessive flushing/writing. By tuning this value, the flush interval can be adjusted to reduce CPU churn.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

- Add the option to the configuration file (`winlogbeat.yml`):
```
winlogbeat.registry_flush: 30s
```
Default is `5s`. Value should be a duration value (i.e., `5s`, `1m`)

## Related issues

- Closes #29001

### CODE REVIEW

File: winlogbeat/beater/winlogbeat.go [hunk 1]
```diff
@@ -110,7 +110,7 @@ func (eb *Winlogbeat) setup(b *beat.Beat) error {
 	config := &eb.config
 
 	var err error
-	eb.checkpoint, err = checkpoint.NewCheckpoint(config.RegistryFile, 10, 5*time.Second)
+	eb.checkpoint, err = checkpoint.NewCheckpoint(config.RegistryFile, 10, config.RegistryFlush)
```
Comment (andrewkroh) [inline winlogbeat/beater/winlogbeat.go:113]: In order to make increasing the value more effective at throttling the number of writes, can you disable the max number of updates parameter (like change the `10` to a `-1` and update the checkpoint package to handle that). Or completely remove that parameter from checkpoint.

**Review by andrewkroh — COMMENTED** [review]
Thanks for adding this. 👍

**Review by andrewkroh — CHANGES_REQUESTED** [review]
The code LGTM. The new configuration option needs to be documented in the Winlogbeat config docs and added to the winlogbeat.reference.yml. Can you please add those two things.

**Comment by taylor-swanson** [comment]
> The code LGTM. The new configuration option needs to be documented in the Winlogbeat config docs and added to the winlogbeat.reference.yml. Can you please add those two things.

Strange, I had added the new config option to the winlogbeat.reference.yml files, but it got reverted somehow in one of my commits. I'll get that added back in, and I'll add to the config docs as well.

**Review by andrewkroh — APPROVED** [review]
LGTM

**Pull request merged**